client/matchmaking/socket-handlers.js

```diff
@@ -90,15 +96,63 @@ const eventToAction = {
     }, 5000)
   },
 
-  matchReady: (name, event) => {
+  matchReady: (name, event) => (dispatch, getState) => {
     dispatch(closeDialog())
-
     clearAcceptMatchTimer()
+
     // All players are ready; feel free to move to the loading screen and start the game
-    return {
+    dispatch({
       type: MATCHMAKING_UPDATE_MATCH_READY,
       payload: event,
+    })
+
+    const {
+      settings,
+      auth: { user },
+    } = getState()
+
+    const {
+      hash,
+      mapData: { format },
+      mapUrl,
+    } = event.mapInfo
+    // NOTE(2Pac): We can't download map any sooner since we don't know which map we'll play until
+    // all players accept the game
+    mapStore.downloadMap(hash, format, mapUrl)
```

Comment:
might wanna file a bug for this, I *believe* my plan was to just download all the matchmaking maps before they could enter the queue (although I guess we still might need to double-check here if they deleted it in the meantime)